fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java

```diff
@@ -324,217 +235,140 @@ public List<Replica> getReplicasByTabletId(long tabletId) {
      * @return list of replica or null if backend not found
      */
     public List<Replica> getReplicasOnBackendByTabletIds(List<Long> tabletIds, long backendId) {
-        readLock();
-        try {
-            Map<Long, Replica> replicaMetaWithBackend = row(backingReplicaMetaTable, backendId);
-            if (!replicaMetaWithBackend.isEmpty()) {
-                List<Replica> replicas = Lists.newArrayList();
-                for (long tabletId : tabletIds) {
-                    replicas.add(replicaMetaWithBackend.get(tabletId));
-                }
-                return replicas;
+        List<Replica> replicas = Lists.newArrayListWithCapacity(tabletIds.size());
+        for (Long tabletId : tabletIds) {
+            Replica replica = getReplica(tabletId, backendId);
+            if (replica != null) {
+                replicas.add(replica);
             }
+        }
+        if (replicas.isEmpty()) {
```

Comment:
the interface contract is changed, original implementation return null only when backend doesn't exist, if the backend exists but doesn't have any replica, an empty list will be returned. After refactor, null will be returned in second case. Need carefully check if the behavior should be changed. If yes, update the comment as well.